### PR TITLE
Refactor several test cases in sqlalchemy store test 

### PR DESCRIPTION
**Pull request by TomeHirata**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/TomeHirata/mlflow/pull/10266?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10266/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10266
```

</p>
</details>

### Related Issues/PRs

https://github.com/mlflow/mlflow/issues/10042

### What changes are proposed in this pull request?

This is a second PR of https://github.com/mlflow/mlflow/issues/10042, which tries to refactor https://github.com/mlflow/mlflow/blob/master/tests/store/tracking/test_sqlalchemy_store.py into the pytest style

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by TomeHirata** [comment]
@harupy Sorry, would you mind taking a look at this PR?

**Comment by TomeHirata** [comment]
@BenWilson2 Would it be possible for you to review this?

File: tests/store/tracking/test_sqlalchemy_store.py
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:1]: /ai please review this

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def _cleanup_database(store: SqlAlchemyStore):
             session.execute(sqlalchemy.sql.text(reset_experiment_id))
 
 
-def _experiment_factory(names, store: SqlAlchemyStore) -> Union[str, List]:
+def _experiment_factory(store: SqlAlchemyStore, names) -> Union[str, List]:
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:197]: nit: `_create_experiments` might be more straightforward.

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -208,6 +208,28 @@ def _experiment_factory(names, store: SqlAlchemyStore) -> Union[str, List]:
     return store.create_experiment(name=names)
 
 
+def _get_run_configs(experiment_id=None, tags=None, start_time=None):
+    return {
+        "experiment_id": experiment_id,
+        "user_id": "Anderson",
+        "start_time": start_time if start_time is not None else get_current_time_millis(),
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:215]:
```suggestion
        "start_time": start_time or get_current_time_millis(),
```

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 3]
```diff
@@ -208,6 +208,28 @@ def _experiment_factory(names, store: SqlAlchemyStore) -> Union[str, List]:
...
+
+
+def _run_factory(store: SqlAlchemyStore, config=None):
+    if not config:
+        config = _get_run_configs()
+
+    experiment_id = config.get("experiment_id", None)
+    if not experiment_id:
+        experiment_id = _experiment_factory(store, "test exp")
+        config["experiment_id"] = experiment_id
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:228]:
```suggestion
    if "experiment_id" not in config:
        config["experiment_id"] = _experiment_factory(store, "test exp")
```

**Review by harupy — APPROVED** [review]
left comments, LGTM once they are addressed!

**Comment by TomeHirata** [comment]
@harupy Thank you for reviewing it. I've addressed the comments, please take a look.

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**